### PR TITLE
Add workflow to update gpuCI

### PR DESCRIPTION
**Pull request by charlesbluca**
Adds a workflow similar to [dask-docker's Dask updating workflow](https://github.com/dask/dask-docker/blob/main/.github/workflows/watch-conda-forge.yml) to update the `RAPIDS_VER` and `UCX_PY_VER` used by gpuCI when a new nightlies of these packages are available.

Currently blocked by some work that needs to be done in https://github.com/jacobtomlinson/gha-anaconda-package-version.

E:

Looks like things resolved themselves - the original issue was that:

- `gha-anaconda-package-version` fails to parse ucx-py nightly versions as valid SemVers due to the tag appended (e.g. `0.23.0a211005`)
- when we set `gha-anaconda-package-version` to assume ucx-py versions are using CalVer, we circumvent this parsing failure, but because there are actually CalVer versions published for ucx-py nightlies, these get picked up instead of the latest SemVer

It looks like the CalVer packages published for ucx-py have been removed, so now the workaround for ucx-py should succeed, but long-term we may want to explore either expanding `gha-anaconda-package-version` to handle versions with tags appended, or use a different method altogether to find the latest versions.

### CODE REVIEW

**Comment by quasiben** [comment]
+1 for automating this.  @jacobtomlinson would you also be willing to review ?

File: .github/workflows/update-gpuci.yml
Comment (jacobtomlinson) [inline .github/workflows/update-gpuci.yml:48]: We might want to switch to `v2` here and use globs instead of regex. Makes this a little cleaner.

**Review by jacobtomlinson — COMMENTED** [review]
This looks great @charlesbluca.

You mention this is blocked on some upstream work in the action. What do we need to do to unblock it?

File: .github/workflows/update-gpuci.yml
Comment (charlesbluca) [reply .github/workflows/update-gpuci.yml:48]: Thanks! I'll also update this in https://github.com/dask/dask/blob/main/.github/workflows/update-gpuci.yml

**Comment by charlesbluca** [comment]
I think the best way of handling these nightly packages from the action would be to add handling for version suffixes - I'm imagining this could be done with an argument providing something like:

- the number of characters to clip off the end of the package version before attempting to parse
- a glob/regex of the nightly tag that we want to exclude when parsing the package version

Right now, things should work by setting `version_system: "CalVer"` to avoid errors parsing the nightly strings; issues would only arise if a CalVer package was accidentally uploaded for ucx-py, as it would immediately take precedent over all SemVer packages. This was what originally caused problems here, but it looks like all ucx-py's packages are SemVer now so we should be safe.

File: .github/workflows/update-gpuci.yml
Comment (charlesbluca) [reply .github/workflows/update-gpuci.yml]: I ended up restructuring the find-replace steps to avoid regex altogether - should make it a little easier to revisit later on

**Review by jacobtomlinson — APPROVED** [review]
This looks great thanks @charlesbluca

**Comment by charlesbluca** [comment]
cc @raydouglass if you want to take a look

**Review by quasiben — APPROVED** [review]

**Pull request merged**
